llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -2083,6 +2086,20 @@ Value *llvm::addRuntimeChecks(
           "stride.check");
       IsConflict = ChkBuilder.CreateOr(IsConflict, IsNegativeStride);
     }
+
+    if (AllChecksHoisted) {
+      AllChecksHoisted &= SE->isLoopInvariant(SE->getSCEV(A.Start), OuterLoop);
```

Comment:
Can we avoid going from IR value -> SCEV and just check the entries in `PointerChecks` for invariance?

Constructing SCEVs here for values that may be removed later may leave dangling IR value entries in the SCEV expression cache?